### PR TITLE
update dependencies and some cleanup

### PR DESCRIPTION
**Pull request by JonathanPrince**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 language: node_js
 node_js:
-  - "4.2"
+  - "8.11"
```
[the remark this answers is not in the document]
Comment (JonathanPrince) [reply .travis.yml]: should probably be `>=`

File: index.js [hunk 2]
```diff
@@ -18,7 +17,7 @@ module.exports = config => {
     }
   }
 
-  config = R.merge(defaults, config)
+  config = Object.assign(defaults, config)
```
[the remark this answers is not in the document]
Comment (JonathanPrince) [reply index.js]: true, probably better form

File: package.json [hunk 3]
```diff
@@ -18,7 +18,7 @@
     "url": "https://mediasuite.co.nz"
   },
   "engines": {
-    "node": ">=4.2.0"
+    "node": ">=8.11.0"
```
[the remark this answers is not in the document]
Comment (JonathanPrince) [reply package.json]: any reason why it wouldn't work with 8?

File: test/adapter-amazon.spec.js [hunk 4]
```diff
@@ -67,44 +61,35 @@ describe('amazon adapter', () => {
...
-      let client = adapter(config)
+      const config = { provider: 'amazon', container: 'dummy' }
+      const client = adapter(config)
 
       // when
-      let download = client.download('my-file.txt', {})
+      const download = client.download('my-file.txt', {})
 
       // then
       download.then(fileStream => {
```
[the remark this answers is not in the document]
Comment (JonathanPrince) [reply test/adapter-amazon.spec.js]: i did consider throwing a promise in there to handle the stream but wasn't convinced, this seemed more explicit than hiding the stream behind a promise

File: package.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (JonForest) [reply package.json]: Nope, just be better to be consistent with the rest of the app was my thinking, but this range is inclusive of 10, so I'm happy.

**Pull request merged**